### PR TITLE
#1-4 Task. Alex Kulik

### PR DESCRIPTION
**Pull request by Senobiot**
This is 1-4 tasks
Link to travis build with passed tests:
https://travis-ci.com/github/Senobiot/js-assignments/builds/174231073
Link to travis build after fix 
https://travis-ci.com/github/Senobiot/js-assignments/builds/174857567

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    return value.match(/(?<=, ).*[^\!]/g)
```
Comment (FedorovD) [inline task/01-strings-tasks.js:72]: should return string, try to solve without regexp

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ function extractNameFromTemplate(value) {
  *   'cat'       => 'c'
  */
 function getFirstChar(value) {
-    throw new Error('Not implemented');
+    return value.match(/^\w/)
```
Comment (FedorovD) [inline task/01-strings-tasks.js:87]: try to solve without regexp

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -99,7 +99,7 @@ function getFirstChar(value) {
  *   '\tHello, World! ' => 'Hello, World!'
  */
 function removeLeadingAndTrailingWhitespaces(value) {
-    throw new Error('Not implemented');
+    return value.replace(/^[^\S]*|[^\S]*$/g, "")
```
Comment (FedorovD) [inline task/01-strings-tasks.js:102]: try to solve without regexp

File: task/01-strings-tasks.js [hunk 4]
```diff
@@ -267,7 +273,13 @@ function isString(value) {
...
 function getCardId(value) {
-    throw new Error('Not implemented');
+    let uni = [9824, 9829, 9830, 65, 74, 75, 81], //diamonds - 9830, pikes - 9824, hearts - 9829, 65 - A, 74 - J, K - 75, Q - 81
+        first = value.charCodeAt(0),
+        sec = value.charCodeAt(value.length - 1),
+        res = 0;
+        first == uni[3] ? res : first == uni[4] ? res = 10 : first == uni[5] ? res = 12 : first == uni[6] ? res = 11 :  res = parseInt(value) - 1;
+        sec == uni[0] ? res += 39 : sec == uni[1] ? res += 26: sec == uni[2] ? res += 13: res;
+    return res; 
 }
```
Comment (FedorovD) [inline task/01-strings-tasks.js:283]: operator `===` is always preferable

File: task/01-strings-tasks.js [hunk 5]
```diff
@@ -267,7 +273,13 @@ function isString(value) {
  *   'K♠' => 51
  */
 function getCardId(value) {
-    throw new Error('Not implemented');
+    let uni = [9824, 9829, 9830, 65, 74, 75, 81], //diamonds - 9830, pikes - 9824, hearts - 9829, 65 - A, 74 - J, K - 75, Q - 81
+        first = value.charCodeAt(0),
+        sec = value.charCodeAt(value.length - 1),
+        res = 0;
+        first == uni[3] ? res : first == uni[4] ? res = 10 : first == uni[5] ? res = 12 : first == uni[6] ? res = 11 :  res = parseInt(value) - 1;
```
Comment (FedorovD) [inline task/01-strings-tasks.js:280]: a bad practice, poorly readable code, try to divide steps

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -415,7 +426,7 @@ function toStringList(arr) {
  *      { country: 'Russia',  city: 'Saint Petersburg' }
  */
 function sortCitiesArray(arr) {
-   throw new Error('Not implemented');
+    return  arr.sort((a,b) => (a.country > b.country) ? 1 : (a.country === b.country) ? ((a.city > b.city) ? 1 : -1) : -1);
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:430]: a bad practice, poorly readable code, try to divide steps

**Comment by FedorovD** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/01-strings-tasks.js (hunk 1 above)
Comment (Senobiot) [reply task/01-strings-tasks.js:72]: Hello. Should I make changes right here using "Edit file"?

**Comment by FedorovD** [comment]
you don't need to use extra spaces in if-else
please follow code style https://javascript.info/coding-style
**Closing the PR as accepted! Thanks!**

**Pull request closed without merging**